packages/devtools_app/lib/src/screens/network/network_controller.dart

```diff
@@ -379,41 +378,43 @@ class NetworkController
   }
 
   @override
-  void filterData(Filter<NetworkRequest> filter) {
+  void filterData(Filter<NetworkRequest>? filter) {
     serviceManager.errorBadgeManager.clearErrors(NetworkScreen.id);
     if (filter?.queryFilter == null) {
       _requests.value.requests.forEach(_checkForError);
       filteredData
         ..clear()
         ..addAll(_requests.value.requests);
     } else {
+      final queryFilter = filter!.queryFilter!;
```

Comment:
if you put this before the if statement above, can we get rid of the second null assertion here?